scRNA-seq/post-processing_scRNA-seq.Rmd

```diff
@@ -0,0 +1,150 @@
+```{r Import Data}
+# Set up function from the COMBINE lab
+ReadAlevin <- function(base.path = NULL){
+  if (! dir.exists(base.path )){
+    stop("Directory provided does not exist")
+  }
+  # Obtain paths to data
+  barcode.loc <- file.path(base.path, "alevin", "quants_mat_rows.txt")
+  gene.loc <- file.path(base.path, "alevin", "quants_mat_cols.txt")
+  matrix.loc <- file.path(base.path, "alevin", "quants_mat.csv")
+    
+  if (!file.exists(barcode.loc)) {
+    stop("Barcode file missing")
+  }
+  if (!file.exists(gene.loc)) {
+    stop("Gene name file missing")
+  }
+  if (!file.exists(matrix.loc)) {
+    stop("Expression matrix file missing")
+  }
+  # Read in the data from alevin output
+  matrix <- as.matrix(read.csv(matrix.loc, header=FALSE))
+  matrix <- t(matrix[,1:ncol(matrix)-1])
+    
+  # Read in the cell and gene names from the alevin output
+  cell.names <- readLines(barcode.loc)
+  gene.names <- readLines(gene.loc)
+    
+  # Apply the colnames and rownames to the dataset
+  colnames(matrix) <- cell.names
+  rownames(matrix) <- gene.names
+    
+  # Make NA values into 0's
+  matrix[is.na(matrix)] <- 0
+  return(matrix)
+}
+
+# Run this function on our files
+alv.data <- ReadAlevin(alevin.file)
+```
+
+#### Set up the Seurat object and apply a filter to the data. 
+In this example, we are applying a gene filter cutoff of at least 3 cells 
+expressing each gene. 
+For cell filtering, each cell must express at least 200 genes.
+```{r Set Up Seurat Object}
+# Set up the seurat object 
+seurat.obj <- CreateSeuratObject(raw.data = alv.data, min.cells = 3,
+                                 min.genes = 200, 
+                                 project = "1k_pbmc_subset")
+```
+
+#### Normalize the data using log normalization
+```{r Normalize the Data}
+seurat.norm <- NormalizeData(object = seurat.obj, 
+                             normalization.method = "LogNormalize",
+                             scale.factor = 10000)
+```
+
+#### Identify variable genes and scale the data
+```{r Find Variable Genes}
+seurat.norm <- FindVariableGenes(object = seurat.norm, mean.function = ExpMean,
+                                x.low.cutoff = 0.0125, x.high.cutoff = 3,
+                                y.cutoff = 0.5)
+
+# Scale the data
+seurat.norm <- ScaleData(object = seurat.norm)
+```
+
+#### Scale data and perform PCA
+```{r Scale Data}
+# Run PCA
+seurat.norm <- RunPCA(object = seurat.norm, pc.genes = seurat.norm@var.genes,
+                     do.print = TRUE, pcs.print = 1:5, genes.print = 5)
+
+# Cluster analyses
+seurat.norm <- FindClusters(object = seurat.norm, reduction.type = "pca", 
+                           dims.use = 1:10, resolution = 0.6, print.output = 0, 
+                           save.SNN = TRUE)
+```
+
+#### Run tSNE on normalized data
+```{r Run tSNE}
+# Run tSNE 
+seurat.norm <- RunTSNE(object = seurat.norm, dims.use = 1:10, do.fast = TRUE)
+
+# Plot the tSNE
+TSNEPlot(object = seurat.norm)
```

Comment:
I think we should manually set the color palette here with `colors.use` to use a scale that is more accessible to folks with color-vision deficiency.